### PR TITLE
[docs] Add integration test example to copy pasta template

### PR DESCRIPTION
**Pull request by sshleifer**
Encourage testing practices that have been encouraged since last update, namely:

- @slow tests that run on cuda and fp16 if possible and show that your model produces good outputs.
- more tests ~= better
- unindent the ModelTester
- call get_extended_attention_mask and delete massive comment.

My code is probably broken because this thing isn't tested!

Add:
"""
Try to make this test take a string and check that a resultant string == desired_result using your tokenizers encode and decode functions
""

### CODE REVIEW

File: templates/adding_a_new_model/tests/test_modeling_xxx.py [hunk 1]
```diff
@@ -268,7 +270,33 @@ def test_for_token_classification(self):
         self.model_tester.create_and_check_xxx_for_token_classification(*config_and_inputs)
 
     @slow
-    def test_model_from_pretrained(self):
```
Comment (sshleifer) [inline templates/adding_a_new_model/tests/test_modeling_xxx.py:271]: this thing is a huge waste of time and money and tests nothing.

File: templates/adding_a_new_example_script/README.md [hunk 2]
```diff
@@ -3,3 +3,5 @@
 This folder provide a template for adding a new example script implementing a training or inference task with the models in the  🤗Transformers library.
 
 Currently only examples for PyTorch are provided which are adaptations of the library's SQuAD examples which implement single-GPU and distributed training with gradient accumulation and mixed-precision (using NVIDIA's apex library) to cover a reasonable range of use cases.
```
Comment (sshleifer) [inline templates/adding_a_new_example_script/README.md:5]: not true

File: templates/adding_a_new_model/tests/test_modeling_xxx.py [hunk 3]
```diff
@@ -31,9 +31,207 @@
...
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.max_position_embeddings = max_position_embeddings
+        self.type_vocab_size = type_vocab_size
+        self.type_sequence_label_size = type_sequence_label_size
```
Comment (sshleifer) [inline templates/adding_a_new_model/tests/test_modeling_xxx.py:83]: would love to not have this take any kwargs./

**Review by sgugger — APPROVED** [review]
Looks good to me, thanks!

Will add another PR since the template does not mention doc at all :-(

File: .github/workflows/self-scheduled.yml [hunk 4]
```diff
@@ -55,4 +55,5 @@ jobs:
         USE_CUDA: yes
       run: |
         source .env/bin/activate
+        pip install -r examples/requirements.txt
```
Comment (sshleifer) [inline .github/workflows/self-scheduled.yml:58]: diff is wrong.

File: templates/adding_a_new_model/tests/test_modeling_xxx.py [hunk 5]
```diff
@@ -31,8 +31,207 @@
...
+        model.to(torch_device)
+        model.eval()
+        loss, logits = model(
+            input_ids, attention_mask=input_mask, token_type_ids=token_type_ids, labels=sequence_labels
+        )
+        result = {
+            "loss": loss,
+            "logits": logits,
+        }
+        self.parent.assertListEqual(list(result["logits"].size()), [self.batch_size, self.num_labels])
```
Comment (sshleifer) [inline templates/adding_a_new_model/tests/test_modeling_xxx.py:204]: ugh

File: templates/adding_a_new_example_script/README.md [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
 This folder provide a template for adding a new example script implementing a training or inference task with the models in the  🤗Transformers library.
+Add tests!
 
-Currently only examples for PyTorch are provided which are adaptations of the library's SQuAD examples which implement single-GPU and distributed training with gradient accumulation and mixed-precision (using NVIDIA's apex library) to cover a reasonable range of use cases.
+
+These folder can be put in a subdirectory under your example's name, like `examples/deebert`.
+
+
+Best Practices: 
+- use `Trainer`
```
Comment (julien-c) [inline templates/adding_a_new_example_script/README.md:11]:
```suggestion
- use `Trainer`/`TFTrainer`
```

File: .github/workflows/self-scheduled.yml (hunk 4 above)
Comment (julien-c) [reply .github/workflows/self-scheduled.yml:58]: ? That's weird

**Review by julien-c — APPROVED** [review]

**Pull request merged**